### PR TITLE
Serialized search index loading is implemented

### PR DESCRIPTION
**Pull request by sgrishchenko**
See main issue [here](https://github.com/TypeStrong/typedoc/issues/1251)

### CODE REVIEW

File: src/default/assets/js/src/typedoc/components/Search.ts [hunk 1]
```diff
@@ -99,77 +100,41 @@ namespace typedoc.search
...
+            const url = this.el.dataset.index;
+            if (!url) {
+                this.setLoadingState(SearchLoadingState.Failure);
+                return;
             }
+
+            fetch(url)
+                .then(response => {
+                    if (!response.ok) {
+                        throw new Error('The source is not found');
```
Comment (Gerrit0) [inline src/default/assets/js/src/typedoc/components/Search.ts]: A bit more specific please, "The search index is missing" would work. "source" is a bit ambiguous.

**Review by Gerrit0 — CHANGES_REQUESTED** [review]
One minor comment, looks good otherwise :)

File: src/default/assets/js/src/typedoc/components/Search.ts (hunk 1 above)
Comment (sgrishchenko) [reply src/default/assets/js/src/typedoc/components/Search.ts]: Done.

**Review by Gerrit0 — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**